packages/protocol/contracts/governance/DoubleSigningSlasher.sol

```diff
@@ -152,9 +149,11 @@ contract DoubleSigningSlasher is Ownable, Initializable, UsingRegistry, UsingPre
     address[] memory groupElectionGreaters,
     uint256[] memory groupElectionIndices
   ) public {
-    require(!isSlashed[keccak256(abi.encodePacked(signer, headerA, headerB))], "Already slashed");
-    uint256 blockNumber = eval(signer, index, headerA, headerB);
-    isSlashed[keccak256(abi.encodePacked(signer, headerA, headerB))] = true;
+    require(!isSlashed[signer][blockHashFromHeader(headerA)], "Already slashed");
```

Comment:
This is definitely an edge case, but if a validator signed blocks A, B, *and* C at the same height, how many times should they be slashed?

In other words, should we:
```
require(!isSlashed[signer][blockHashFromHeader(headerA)] || !isSlashed[signer][blockHashFromHeader(headerB)]);
```

or:
```
require(!isSlashed[signer][blockHashFromHeader(headerA)] && !isSlashed[signer][blockHashFromHeader(headerB)]);
```